lib/internal/Magento/Framework/App/View/Asset/Publisher.php

```diff
@@ -42,14 +42,33 @@ public function __construct(
      */
     public function publish(Asset\LocalInterface $asset)
     {
-        $dir = $this->filesystem->getDirectoryRead(DirectoryList::STATIC_VIEW);
-        if ($dir->isExist($asset->getPath())) {
+        if ($this->isFileEquals($asset)) {
             return true;
         }
 
         return $this->publishAsset($asset);
     }
 
+    public function isFileEquals(Asset\LocalInterface $asset)
+    {
+        $dir = $this->filesystem->getDirectoryRead(DirectoryList::STATIC_VIEW);
+        if ($dir->isExist($asset->getPath()) === false) {
+            return false;
+        }
+
+        $source = $asset->getSourceFile();
+        $destination = $dir->getAbsolutePath($asset->getPath());
+
+        $sourceSum = md5_file($source);
+        $destinationSum = md5_file($destination);
+
+        if ($sourceSum !== $destinationSum) {
+            return false;
+        }
+
+        return true;
+    }
```

Comment:
I actually I don't think shortening it will be best. In the future the `md5_file()` lines might need to be replaced with a wrapped class call (currently `$this->filesystem` does not support MD5 calls). So I think the explicit call consumes more lines indeed, but allows for better modification in the future.